packages/filesystem/src/browser/download/file-download-service.ts

```diff
@@ -147,6 +147,39 @@ export class FileDownloadService {
         return this.deferredUpload.promise;
     }
 
+    async copyDownloadLink(uris: URI[]): Promise<void> {
+        if (uris.length === 0) {
+            return;
+        }
+        try {
+            await this.statusBar.setElement(FileDownloadService.PREPARING_DOWNLOAD_ID, {
+                alignment: StatusBarAlignment.RIGHT,
+                text: '$(spinner~spin) Preparing download link...',
+                tooltip: 'Preparing download link...',
+                priority: 1
+            });
+            const response = await fetch(this.request(uris));
+            const jsonResponse = await response.json();
+            const { status, statusText } = response;
+            await this.statusBar.removeElement(FileDownloadService.PREPARING_DOWNLOAD_ID);
+            if (status === 200) {
+                const url = `${this.endpoint()}/download/?id=${jsonResponse.id}`;
+                const setClipBoardData = (event: ClipboardEvent) => {
+                    event.clipboardData.setData('text/plain', url);
+                    event.preventDefault();
+                };
+                document.addEventListener('copy', setClipBoardData);
+                document.execCommand('copy', false);
```

Comment:
Using Firefox 66.0.2 (64-bit) on Linux, this line doesn't actually trigger the clipboard copy callback :(

Log I see appearing in Firefox's console:
```
document.execCommand(‘cut’/‘copy’) was denied because it was not called from inside a short running user-generated event handler.
```

See https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard#Using_execCommand()

Reply:
Thanks for the review, i'll check it out :)